packages/react-dom/src/events/BrowserEventConstants.js

```diff
@@ -3,95 +3,143 @@
  *
  * This source code is licensed under the MIT license found in the
  * LICENSE file in the root directory of this source tree.
+ *
+ * @flow
  */
 
+import type {TopLevelType} from 'events/TopLevelEventTypes';
+import invariant from 'fbjs/lib/invariant';
+
+import * as DOMTopLevelEventTypes from './DOMTopLevelEventTypes';
 import getVendorPrefixedEventName from './getVendorPrefixedEventName';
 
+/**
+ * Small helper function to create a map using the Map constructor arguments.
+ * This is needed because IE11 does not implement that constructor.
+ */
+function makeMap<K, V>(values: Array<[K, V]>): Map<K, V> {
+  const map: Map<K, V> = new Map();
+  values.forEach(([k, v]) => {
+    map.set(k, v);
+  });
+  return map;
+}
```

Comment:
If we're getting to the point where we have to introduce meta-programming to get around this IE11 Map issue, we should seriously consider requiring a polyfill in the next major release.